Dockerfile

```diff
@@ -35,8 +35,12 @@ RUN apt-get update && apt-get install -y \
 
 # Install rust using rustup
 ARG CHANNEL="nightly"
-RUN curl https://sh.rustup.rs -sSf | \
-    sh -s -- -y --default-toolchain ${CHANNEL} --profile minimal && \
+ENV RUSTUP_VER="1.20.2" \
+    RUST_ARCH="x86_64-unknown-linux-gnu"
+RUN curl "https://static.rust-lang.org/rustup/archive/${RUSTUP_VER}/${RUST_ARCH}/rustup-init" -o rustup-init && \
+    chmod +x rustup-init && \
+    ./rustup-init -y --default-toolchain ${CHANNEL} --profile minimal && \
+    rm rustup-init && \
```

Comment:
any particular reason why you saving this temporary script and not just piping it to `sh` ?

Reply:
It's an executable, which I don't think you can pipe.

Reply:
oh, wow it's a big binary. okay, makes sense then.